test/functional/p2p-compactblocks.py

```diff
@@ -43,51 +35,33 @@ def on_close(self, conn):
     def on_sendcmpct(self, conn, message):
         self.last_sendcmpct.append(message)
 
-    def on_block(self, conn, message):
-        self.last_block = message
-
     def on_cmpctblock(self, conn, message):
-        self.last_cmpctblock = message
         self.block_announced = True
-        self.last_cmpctblock.header_and_shortids.header.calc_sha256()
-        self.set_announced_blockhashes.add(self.last_cmpctblock.header_and_shortids.header.sha256)
+        self.last_message["cmpctblock"].header_and_shortids.header.calc_sha256()
+        self.set_announced_blockhashes.add(self.last_message["cmpctblock"].header_and_shortids.header.sha256)
```

Comment:
Please let me know if this is inappropriate, but the object attribute `set_announced_blockhashes` is a bit confusing. Would changing the name to something like `announced_blockhashes_set` be appropriate here?